chall04/ptuukkan.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/python3
+# Version 3.6.8
+
+import sys
+
+def validate_input(lines):
+	if len(lines) == 0:
+		return 0
+	dim = lines.pop(0)
+	if not dim.isnumeric():
+		return 0
+	dim = int(dim)
+	if len(lines) != dim:
+		return 0
+	for line in lines:
+		if len(line) != dim:
+			return 0
+		for c in line:
+			if not c in " .#":
```

Comment:
`if set(line) > set('#. '):`
set is very useful data type, I suggest learning it